File: OffByOne/src/OffByOne.Ql/Values/IntegerValue.cs
namespace OffByOne.Ql.Values
{
    using System.Globalization;

    using MoreDotNet.Extensions.Common;

    using OffByOne.Ql.Values.Base;
    using OffByOne.Ql.Values.Contracts;

    public class IntegerValue : BaseValue
    {
        public IntegerValue(string value)
            : this(int.Parse(value, CultureInfo.InvariantCulture))
        {
        }

        public IntegerValue(int value)
        {
            this.Value = value;
        }

        public int Value { get; }

        public override IValue Parse(string value)
        {
            return new IntegerValue(value);
        }

        public override IValue Add(IValue other)
        {
            return other.Add(this);
        }

        public override IValue Add(DecimalValue other)
        {
            return new DecimalValue(other.Value + this.Value);
        }

        public override IValue Add(MoneyValue other)
        {
            return new MoneyValue(other.Value + this.Value);
        }

        public override IValue Add(IntegerValue other)
        {
            return new IntegerValue(other.Value + this.Value);
        }

        public override IValue Add(StringValue other)
        {
            return new StringValue(other.Value + this.Value.ToString(CultureInfo.InvariantCulture));
        }

        public override IValue Substract(IValue other)
        {
            return other.Substract(this);
        }

        public override IValue Substract(DecimalValue other)
        {
            return new DecimalValue(other.Value - this.Value);
        }

        public override IValue Substract(MoneyValue other)
        {
            return new MoneyValue(other.Value - this.Value);
        }

        public override IValue Substract(IntegerValue other)
        {
            return new IntegerValue(other.Value - this.Value);
        }

        public override IValue Divide(IValue other)
        {
            return other.Divide(this);
        }

        public override IValue Divide(DecimalValue other)
        {
            return new DecimalValue(other.Value / this.Value);
        }

        public override IValue Divide(MoneyValue other)
        {
            return new MoneyValue(other.Value / this.Value);
        }

        public override IValue Divide(IntegerValue other)
        {
            return new IntegerValue(other.Value / this.Value);
        }

        public override IValue Multiply(IValue other)
        {
            return other.Multiply(this);
        }

        public override IValue Multiply(DecimalValue other)
        {
            return new DecimalValue(other.Value * this.Value);
        }

        public override IValue Multiply(MoneyValue other)
        {
            return new MoneyValue(other.Value * this.Value);
        }

        public override IValue Multiply(IntegerValue other)
        {
            return new IntegerValue(other.Value * this.Value);
        }

        public override BooleanValue LessThan(IValue other)
        {
            return other.LessThan(this);
        }

        public override BooleanValue LessThan(DecimalValue other)
        {
            return new BooleanValue(other.Value < this.Value);
        }

        public override BooleanValue LessThan(MoneyValue other)
        {
            return new BooleanValue(other.Value < this.Value);
        }

        public override BooleanValue LessThan(IntegerValue other)
        {
            return new BooleanValue(other.Value < this.Value);
        }

        public override BooleanValue GreaterThan(IValue other)
        {
            return other.GreaterThan(this);
        }

        public override BooleanValue GreaterThan(DecimalValue other)
        {
            return new BooleanValue(other.Value > this.Value);
        }

        public override BooleanValue GreaterThan(MoneyValue other)
        {
            return new BooleanValue(other.Value > this.Value);
        }

        public override BooleanValue GreaterThan(IntegerValue other)
        {
            return new BooleanValue(other.Value > this.Value);
        }

        public override BooleanValue GreaterThanOrEqualTo(IValue other)
        {
            return other.GreaterThanOrEqualTo(this);
        }

        public override BooleanValue GreaterThanOrEqualTo(MoneyValue other)
        {
            return new BooleanValue(other.Value >= this.Value);
        }

        public override BooleanValue GreaterThanOrEqualTo(IntegerValue other)
        {
            return new BooleanValue(other.Value >= this.Value);
        }

        public override BooleanValue GreaterThanOrEqualTo(DecimalValue other)
        {
            return new BooleanValue(other.Value >= this.Value);
        }

        public override BooleanValue LessThanOrEqualTo(IValue other)
        {
            return other.LessThanOrEqualTo(this);
        }

        public override BooleanValue LessThanOrEqualTo(DecimalValue other)
        {
            return new BooleanValue(other.Value <= this.Value);
        }

        public override BooleanValue LessThanOrEqualTo(MoneyValue other)
        {
            return new BooleanValue(other.Value <= this.Value);
        }

        public override BooleanValue LessThanOrEqualTo(IntegerValue other)
        {
            return new BooleanValue(other.Value <= this.Value);
        }

        public override BooleanValue Equals(IValue other)
        {
            return other.Equals(this);
        }

        public override BooleanValue Equals(IntegerValue other)
        {
            return new BooleanValue(other.Value == this.Value);
        }

        public override BooleanValue Equals(DecimalValue other)
        {
            return new BooleanValue(other.Value == this.Value);
        }

        public override BooleanValue Equals(MoneyValue other)
        {
            return new BooleanValue(other.Value == this.Value);
        }

        public override IValue Negative()
        {
            return new IntegerValue(-this.Value);
        }

        public override IValue Positive()
        {
            return new IntegerValue(+this.Value);
        }

        public override string ToString()
        {
            return this.Value.ToString(CultureInfo.InvariantCulture);
        }

        public override int GetHashCode()
        {
            return this.Value.GetHashCode();
        }

        public override bool Equals(object obj)
        {
            return obj.Is<IntegerValue>() && obj.As<IntegerValue>().Value == this.Value;
        }
    }
}

File: vanVlaanderenDerksen/QLJava/src/sc/ql/gui/values/IntegerValue.java
package sc.ql.gui.values;

public class IntegerValue extends Value {
	private final Integer value;
	
	public IntegerValue(Integer value) {
		this.value = value;
	}
	
	@Override
	public Integer toInteger() {
		return value;
	}
	
	@Override
	public String toString() {
		return value.toString();
	}
	
	@Override
	public Value add(Value value) {
        return value.add(this);
    }
	
	@Override
	public IntegerValue add(IntegerValue value) {
        return new IntegerValue(this.value + value.toInteger());
    }
	
	@Override
	public Value divide(Value value) {
        return value.divide(this);
    }
	
	@Override
	public IntegerValue divide(IntegerValue value) {
        return new IntegerValue(this.value / value.toInteger());
    }
	
	@Override
	public Value multiply(Value value) {
        return value.multiply(this);
    }
	
	@Override
	public IntegerValue multiply(IntegerValue value) {
        return new IntegerValue(this.value * value.toInteger());
    }
	
	@Override
	public Value subtract(Value value) {
        return value.subtract(this);
    }
	
	@Override
	public IntegerValue subtract(IntegerValue value) {
        return new IntegerValue(this.value - value.toInteger());
    }
	
	@Override
	public Value equals(Value value) {
        return value.equals(this);
    }
	
	@Override
	public BooleanValue equals(IntegerValue value) {
        return new BooleanValue(value.toInteger() == this.value);
    }
	
	@Override
	public Value equalsNot(Value value) {
        return value.equalsNot(this);
    }
	
	@Override
	public BooleanValue equalsNot(IntegerValue value) {
        return new BooleanValue(value.toInteger() != this.value);
    }
	
	@Override
	public Value greaterThen(Value value) {
        return value.greaterThen(this);
    }
	
	@Override
	public BooleanValue greaterThen(IntegerValue value) {
        return new BooleanValue(this.value > value.toInteger());
    }
	
	@Override
	public Value greaterThenEqual(Value value) {
        return value.greaterThenEqual(this);
    }
	
	@Override
	public BooleanValue greaterThenEqual(IntegerValue value) {
        return new BooleanValue(this.value >= value.toInteger());
    }
	
	@Override
	public Value lessThen(Value value) {
        return value.lessThen(this);
    }
	
	@Override
	public BooleanValue lessThen(IntegerValue value) {
        return new BooleanValue(this.value < value.toInteger());
    }
	
	@Override
	public Value lessThenEqual(Value value) {
        return value.lessThenEqual(this);
    }
	
	@Override
	public BooleanValue lessThenEqual(IntegerValue value) {
        return new BooleanValue(this.value <= value.toInteger());
    }
	
}
